app/main.go

```diff
@@ -247,6 +249,8 @@ func Main(config Config) {
 		if err != nil {
 			log.Fatalf("failed to open environment: %s", err)
 		}
+		// set the environment value for further propagation.
+		env.RequireDigests = config.RequireDigests
```

Comment:
This shouldn't be a public variable. It would be better to pass the setting through the `Env` and `State` constructors, and drop the extra parameter to `UpgradChannel()`.

Reply:
good point. changed.